python/whylogs/core/metrics/condition_count_metric.py

```diff
@@ -15,52 +13,9 @@
 logger = logging.getLogger(__name__)
 
 
-class Relation(Enum):
-    match = 1
-    fullmatch = 2
-    equal = 3
-    less = 4
-    leq = 5
-    greater = 6
-    geq = 7
-    neq = 8
-
-
-def relation(op: Relation, value: Union[str, int, float]) -> Callable[[Any], bool]:
-    if op == Relation.match:
-        return lambda x: re.compile(value).match(x)  # type: ignore
-    if op == Relation.fullmatch:
-        return lambda x: re.compile(value).fullmatch(x)  # type: ignore
-    if op == Relation.equal:
-        return lambda x: x == value  # type: ignore
-    if op == Relation.less:
-        return lambda x: x < value  # type: ignore
-    if op == Relation.leq:
-        return lambda x: x <= value  # type: ignore
-    if op == Relation.greater:
-        return lambda x: x > value  # type: ignore
-    if op == Relation.geq:
-        return lambda x: x >= value  # type: ignore
-    if op == Relation.neq:
-        return lambda x: x != value  # type: ignore
-    raise ValueError("Unknown ConditionCountMetric predicate")
-
-
-def and_relations(left: Callable[[Any], bool], right: Callable[[Any], bool]) -> Callable[[Any], bool]:
-    return lambda x: left(x) and right(x)
-
-
-def or_relations(left: Callable[[Any], bool], right: Callable[[Any], bool]) -> Callable[[Any], bool]:
-    return lambda x: left(x) or right(x)
-
-
-def not_relation(relation: Callable[[Any], bool]) -> Callable[[Any], bool]:
-    return lambda x: not relation(x)
-
-
 @dataclass(frozen=True)
 class Condition:
-    relation: Callable[[Any], bool]
+    relation: Tuple[Callable[[Any, Callable[[], Any]], bool], Callable[[], Union[str, int, float]]]
```

Comment:
Oh wow. THat is quite the datatype. I think it's fine for now, but this looks like a code smell that at the point this many layers... we should probably think of a data structure to make this clearer and easier to read/maintain.